### ISSUE TITLE
Training with IPA + English chars

### ISSUE THREAD

**Issue by lumpidu** [SWivid/F5-TTS#612]
### Checks

- [X] This template is only for question, not feature requests or bug reports.
- [X] I have thoroughly reviewed the project documentation and read the related paper(s).
- [X] I have searched for existing issues, including closed ones, no similar questions.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### Question details

if I'd fine-tune the base model for a new language and at the same time also integrate English data, could I use `IPA` phones for the new language and use additionally normal text as before for the English training part ?
The idea is to have more control over pronunciation of the target language while preserving the ability to speak English and to guide the model better, when to apply which code.

I have seen, that `vocab.txt` already contains most of the `IPA` phone-set (but not all IPA characters are integrated and also no secondary stress symbol).

I am confused about the meaning of the used vocabulary **for the trained base-model**:

- what is the meaning of `vocab.txt` ?
- is one free to choose any UTF-8 character for training/finetuning or only what is inside `vocab.txt` ?
- what is the consequence of keeping the character sets of already trained/ to be finetuned datasets separate ? Does this help code-switching, e.g. for English/Chinese ?
- have you already tried using phones instead of normalized text ? This could also simply be done via `SAMPA`, if `IPA` symbols cannot be supported by the base-model

**Comment by ZhikangNiu** [comment]
1. The vocab contains character tokens (for modeling English) and pinyin initials and finals (for modeling Chinese).
2. You can freely use any UTF-8 characters.
3. This means you need to retrain the word embeddings.
4. We have tried using IPA, and the results were also very good.

**Comment by ZhikangNiu** [comment]
3）Why would this approach help improve code-switching? Could you provide a detailed example? However, this approach might reduce your training costs.

4）Using phonemes results in a smaller performance drop for English, but a larger performance drop for Chinese, as it cannot effectively handle polyphonic characters (characters with multiple pronunciations).

**Comment by ZhikangNiu** [comment]
3. Indeed, this approach can preserve the ability for code-switching.
4. The performance in English doesn't degrade significantly, but the main issue is a considerable decline in Chinese performance, primarily due to the problem of polyphonic characters (the same character having different pronunciations).

**Comment by lbdave94** [comment]
@lumpidu Hi,

at the end did you try to fine-tuned the model using phonemes? If yes, what are your results?
